metabase/metabase
# Custom homepage

Closes https://github.com/metabase/metabase/issues/30541
Closes https://github.com/metabase/metabase/issues/30542
Closes https://github.com/metabase/metabase/issues/30543

### Description
MVP for pinning a dashboard as a homepage. This PR includes updates to the admin section, permission checks as part of the `/api/current` route, and the edit button CTA for admins on the home page

### How to verify
Pull down a run the branch

1. Admin -> General
2. Enable Custom Homepage, and select a dashboard from the selector
3. Exit admin. You should be brough to the dashboard you selected.

Now, anytime anyone with access to that dashboard logs in (or goes to the home url `/`) will be redirected to the selected dashboard. Anyone without access to that dashboard will have the existing experience with the default home screen.

Alternatively, this property can be set using the customize button on the home screen as shown in the demo below

### Demo
![chrome_acTKK0Govv](https://user-images.githubusercontent.com/1328979/236031047-256258d6-598b-4ca3-9036-54227bc3ec6d.gif)

### Checklist


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30545)
